src/sage/graphs/base/static_dense_graph.pyx

```diff
@@ -378,7 +379,371 @@ def triangles_count(G):
+    EXAMPLES::
+
+        sage: from sage.graphs.base.static_dense_graph import connected_full_subgraphs
+        sage: G = digraphs.Complete(3)
+        sage: len(list(connected_full_subgraphs(G)))
+        54
+    """
+    if not edges:
+        return
+    L = []
+    for u, v in edges:
+        tmp = []
+        if G.has_edge(u, v):
+            tmp.append([(u, v)])
+        if G.has_edge(v, u):
+            tmp.append([(v, u)])
+            if G.has_edge(u, v):
+                tmp.append([(u, v), (v, u)])
+        L.append(tmp)
+
+    if len(L) == 1:
+        for F in L[0]:
+            if min_edges <= len(F) and len(F) <= max_edges:
+                yield _format_result(G, F, edges_only, labels)
+    else:
+        for E in product(*L):
+            F = [e for le in E for e in le]
+            if min_edges <= len(F) and len(F) <= max_edges:
+                yield _format_result(G, F, edges_only, labels)
+    return
+
+
+def connected_full_subgraphs(G, edges_only=False, labels=False,
+                             min_edges=None, max_edges=None):
+    r"""
+    Iterator over the connected subgraphs of `G` with same vertex set.
+
+    This method implements a iterator over the connected subgraphs of the input
+    (di)graph with the same ground set of vertices. That is, it iterates over
+    every subgraph `H = (V_H, E_H)` of `G = (V, E)` such that `V_H = V`,
+    `E_H \subseteq E` and `H` is connected. Hence, this method may yield a huge
+    number of graphs.
+
+    When the input (di)graph `G` is not connected, this method returns nothing.
+
+    As for method :meth:`sage.graphs.generic_graph.connected_components`, edge
+    orientation is ignored. Hence, the directed graph with a single arc `0 \to
+    1` is considered connected.
+
+    INPUT:
+
+    - ``G`` -- a :class:`Graph` or a :class:`DiGraph`; loops and multiple edges
+      are *not* allowed
+
+    - ``edges_only`` -- boolean (default: ``False``); whether to return
+      (Di)Graph or list of vertices
+
+    - ``labels`` -- boolean (default: ``False``); whether to return labelled
+      edges or not. This parameter is used only when ``edges_only`` is ``True``.
+
+    - ``min_edges`` -- integer (default: ``None``); minimum number of edges of
+      reported subgraphs. By default (``None``), this lower bound will be set to
+      `n - 1`.
+
+    - ``max_edges`` -- integer (default: ``None``); maximum number of edges of
+      reported subgraphs. By default (``None``), this lower bound will be set to
+      the number of edges of the input (di)graph.
+
+    .. NOTE::
+
+        Roughly, this method explores all possible subsets of neighbors of each
+        vertex, which represents a huge number of subsets. We have thus chosen
+        to limit the degree of the vertices of the graphs that can be
+        considered, even if the graph has a single connected subgraph (e.g., a
+        tree). This can certainly be improved using a decomposition into
+        biconnected components.
+
```

Comment:
When I first wrote this method, I decided to postpone the preprocessing to a futur PR as it was already a complex code to review.
But since you asked for it, I did it.